Sources/NIOWebSocket/WebSocketFrame.swift

```diff
@@ -0,0 +1,269 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//
+//  WebSocketFrame.swift
+//  NIOWebSocket
+//
+//  Created by Cory Benfield on 16/02/2018.
+//
+
+import NIO
+
+private extension UInt8 {
+    func bitSetWithMask(_ mask: UInt8) -> Bool {
+        return self & mask != 0
+    }
+
+    mutating func setBitForMask(_ mask: UInt8, to: Bool) {
```

Comment:
given that this is mutating I think the Swift conventions say it should be `setting...`. But `setting` might also be misleading as you might also `unset` them, right? What about `changingBitsInMask(_ mask: UInt8, to: Bool)`?

Reply:
other question would be if you really want to use this as the mask instead of 'calculating' the mask here and taking a bit index?

Reply:
I don't think it really matters? It's a purely internal method, so I don't think I care. Whatever you'd prefer.

Reply:
it should definitely be an `ing` method so I'll defer to you if you want to use a mask or a bit index. So I'd argue for `changingBit(index: Int, to: Bool)` or `changingBitsInMask(_ mask: UInt8 to: Bool)`